jiant/tasks/lib/qasrl.py

```diff
@@ -0,0 +1,81 @@
+import gzip
+import nltk
+import json
+
+from jiant.tasks.lib.templates import span_prediction as span_pred_template
+from jiant.utils.retokenize import TokenAligner
+
+
+class QASRLTask(span_pred_template.AbstractSpanPredicationTask):
+    def get_train_examples(self):
+        return self._create_examples(self.train_path, set_type="train")
+
+    def get_val_examples(self):
+        return self._create_examples(self.val_path, set_type="val")
+
+    def get_test_examples(self):
+        return self._create_examples(self.test_path, set_type="test")
+
+    def _create_examples(self, file_path, set_type):
+
+        with gzip.open(file_path) as f:
+            lines = f.read().splitlines()
+
+        examples = []
+        ptb_detokenizer = nltk.tokenize.treebank.TreebankWordDetokenizer()
+
+        for line in lines:
+            datum = json.loads(line)
+            datum = {
+                "sentence_tokens": datum["sentenceTokens"],
+                "entries": [
+                    {
+                        "verb": verb_entry["verbInflectedForms"]["stem"],
+                        "verb_idx": verb_idx,
+                        "questions": {
+                            question: [
+                                [
+                                    {
+                                        "tokens": datum["sentenceTokens"][span[0] : span[1] + 1],
+                                        "span": span,
+                                    }
+                                    for span in answer_judgment["spans"]
+                                ]
+                                for answer_judgment in q_data["answerJudgments"]
+                                if answer_judgment["isValid"]
+                            ]
+                            for question, q_data in verb_entry["questionLabels"].items()
+                        },
+                    }
+                    for verb_idx, verb_entry in datum["verbEntries"].items()
+                ],
+            }
+
+            passage_ptb_tokens = datum["sentence_tokens"]
+            passage_space_tokens = ptb_detokenizer.detokenize(passage_ptb_tokens).split()
+            passage_space_str = " ".join(passage_space_tokens)
+
+            token_aligner = TokenAligner(source=passage_ptb_tokens, target=passage_space_tokens)
+
+            for entry in datum["entries"]:
+                for question, answer_list in entry["questions"].items():
+                    for answer in answer_list:
+                        for answer_span in answer:
+                            answer_char_span = token_aligner.project_token_to_char_span(
+                                answer_span["span"][0], answer_span["span"][1], inclusive=True
```

Comment:
Connected to the comment above — it looks like treating the span as inclusive will overshoot the intended span.